client/state/sites/launch/actions.js

```diff
@@ -15,22 +15,25 @@ export const launchSite = ( siteId ) => ( {
 	},
 } );
 
-export const launchSiteOrRedirectToLaunchSignupFlow = ( siteId ) => ( dispatch, getState ) => {
-	if ( ! isUnlaunchedSite( getState(), siteId ) ) {
-		return;
-	}
+export const launchSiteOrRedirectToLaunchSignupFlow =
+	( siteId, source = null ) =>
+	( dispatch, getState ) => {
+		if ( ! isUnlaunchedSite( getState(), siteId ) ) {
+			return;
+		}
 
-	const isAnchorPodcast = getSiteOption( getState(), siteId, 'anchor_podcast' );
-	const isPaidWithDomain =
-		isCurrentPlanPaid( getState(), siteId ) && getDomainsBySiteId( getState(), siteId ).length > 1;
+		const isAnchorPodcast = getSiteOption( getState(), siteId, 'anchor_podcast' );
+		const isPaidWithDomain =
+			isCurrentPlanPaid( getState(), siteId ) &&
+			getDomainsBySiteId( getState(), siteId ).length > 1;
 
-	if ( isPaidWithDomain || isAnchorPodcast ) {
-		dispatch( launchSite( siteId ) );
-		return;
-	}
+		if ( isPaidWithDomain || isAnchorPodcast ) {
+			dispatch( launchSite( siteId ) );
+			return;
+		}
 
-	const siteSlug = getSiteSlug( getState(), siteId );
+		const siteSlug = getSiteSlug( getState(), siteId );
 
-	// TODO: consider using the `page` library instead of calling using `location.href` here
-	window.location.href = `/start/launch-site?siteSlug=${ siteSlug }`;
-};
+		// TODO: consider using the `page` library instead of calling using `location.href` here
+		window.location.href = `/start/launch-site?siteSlug=${ siteSlug }&source=${ source }`;
```

Comment:
If `source` is `null` (which is the default param value) then `&source=null` gets appended to the string i.e. the `${}` syntax doesn't stringify `null` values to empty strings like this code seems to assume.

This is a nitpick because (1) `launchSiteOrRedirectToLaunchSignupFlow` is currently only ever used in a place that also provides an explicit `source` param, and (2) `?source=null` doesn't cause any issues in the domain step. It just feels a bit untidy.

One option is to use [the `addQueryArgs()` helper function](https://github.com/Automattic/wp-calypso/blob/edbc4c0b02e6252d59b0dc01a909eb3ce3600023/client/lib/url/add-query-args.ts#L12) which deals with nullish values by removing them.

```suggestion
		window.location.href = addQueryArgs( { siteSlug, source }, '/start/launch-site' );
```

Reply:
> One option is to use [the addQueryArgs() helper function](https://github.com/Automattic/wp-calypso/blob/edbc4c0b02e6252d59b0dc01a909eb3ce3600023/client/lib/url/add-query-args.ts#L12) which deals with nullish values by removing them.

Good suggestion! Added with 2aceaa9817